app/server/methods/homes.js

```diff
@@ -223,17 +223,26 @@ Meteor.methods({
     return { activityCounts, inactivityCounts, semiActivityCounts, date };
   },
   getHomeSelectOptionsWithGroups() {
-    // Get all Groups
-    var groups = Groups.find().fetch();
+    const userId = Meteor.userId();
+    const userIsAdmin = Roles.userIsInRole(userId, "admin");
+    let allowedGroups = []
+    if(!userIsAdmin){
```

Comment:
Again, this code is a bit claustrophobic. Just give it a bit of breathing room with whitespace, like paragraph breaks.